peachjam/views/documents.py

```diff
@@ -86,7 +86,14 @@ def render_to_response(self, context, **response_kwargs):
 class DocumentSourcePDFView(DocumentSourceView):
     def render_to_response(self, context, **response_kwargs):
         if hasattr(self.object, "source_file"):
-            file = self.object.source_file.as_pdf()
+            source_file = self.object.source_file
+            file = source_file.as_pdf()
+
+            # redirect search engine crawlers to the original source files
+            # especially for gazettes
+            if source_file.source_url:
```

Comment:
I think we should only do this for DocumentSourceView, since that is the view that returns the original source file, no matter its format. This view is explicitly about converting it to PDF, and we don't know if the source URL will provide a PDF.

Reply:
@longhotsummer  What happens when the crawlers hit `source.pdf`? Will they not force the file to be downloaded from s3 to be converted?

Reply:
Good question. We only advertise a source.pdf link for non-PDF documents. So I think we're okay on this for now.

Reply:
@longhotsummer We are also using `source.pdf` to load source files in the detail view if they are pdfs i.e that is what pdf.js is using to render the pdfs